packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -114,6 +114,12 @@ class ColdRunner extends ResidentRunner {
 
     appStartedCompleter?.complete();
 
+    if (debuggingOptions.vmserviceOutFile != null) {
+      final File vmserviceOutFile = fs.file(debuggingOptions.vmserviceOutFile);
+      vmserviceOutFile.createSync(recursive: true);
```

Comment:
Should we catch exceptions if creating the file fails?

Reply:
Yes!

Reply:
Fixed by moving into base class, adding a catch and a new test case